odoo/addons/base/models/ir_cron.py

```diff
@@ -291,84 +304,156 @@ def _acquire_one_job(cls, cr, job_ids):
+        In case it notify that not all records could have been processed,
+        the cron will be executed again as soon as possible and process the remaining records.
+        When the section action reported that all records have been processed,
+        the cron is scheduled again in the future using its interval configuration.
+
+        In case all executions of the cron fail in a period of {MIN_DELTA_BEFORE_DEACTIVATION}
+        and that there are at least {MIN_FAILURES_BEFORE_DEACTIVATION} consecutive failures then the cron is
+        automatically deactivated and the admin notified."""
+
+        # `failed` indicates the current cron raised an error and did not progress (logged on the current progress)
+        # `timed_out` indicates the last cron call timed_out and did not progress (logged on the last progress)
+        # CRON run up to MAX_BATCH_PER_CRON_CALL times to reduce transaction size as long as there is progress and records remaining
+        # If there is still remaining records, the cron will be re-scheduled
+        # CRON which fail MIN_FAILURES_BEFORE_DEACTIVATION times in a row are deactivated if the first fail is at least MIN_DELTA_BEFORE_DEACTIVATION old
+
+        env = api.Environment(cron_cr, job['user_id'], {})
+        ir_cron = env[cls._name]
+        now = fields.Datetime.context_timestamp(ir_cron, datetime.utcnow())
+
+        timed_out_counter = 0 if job.get('done', 0) == 0 else (job.get('timed_out', 0) or 0)
+        if timed_out_counter < CONSECUTIVE_TIMEOUT_FOR_FAILURE:
+            failed, completed = cls._execute_job(job, timed_out_counter)
+        else:
+            failed = True       # A timed-out job is failed
+            completed = True    # A timed-out job is not re-triggered
+            cron_cr.execute("""
+                UPDATE ir_cron_progress
+                SET timed_out = 'f'
+                WHERE id = %s
+            """, job['progress_id'])
+            _logger.exception('Job `%s` (%s): Server action #%s timed-out.',
+                              job['cron_name'], job['id'], job['ir_actions_server_id'])
+
+        if failed:
+            # Deactivate the cron if it failed 5 times in a row and the first fail is at least 7 days old
+            deactivate = (
+                job['failures'] >= MIN_FAILURES_BEFORE_DEACTIVATION
+                and job['first_failure_date']
+                and fields.Datetime.context_timestamp(ir_cron, job['first_failure_date']) + MIN_DELTA_BEFORE_DEACTIVATION < now
+            )
+            if deactivate:
+                job['active'] = False
+                job['failures'] = 0
+                job['first_failure_date'] = None
+                ir_cron._notify_admin_deactivation(job['cron_name'], job['id'], job['ir_actions_server_id'])
+            else:
+                job['failures'] = job['failures'] + 1
+                job['first_failure_date'] = job['first_failure_date'] or now
```

Comment:
The condition on `job['failures']` is incorrect: because it hasn't been incremented yet, you will need `MIN_FAILURES_BEFORE_DEACTIVATION + 1` failures in a row before deactivation.

I suggest to always increment `failures` first, and then test for deactivation:
```suggestion
            job['failures'] += 1
            job['first_failure_date'] = job['first_failure_date'] or now
            if (
                job['failures'] >= MIN_FAILURES_BEFORE_DEACTIVATION
                and job['first_failure_date']
                and fields.Datetime.context_timestamp(ir_cron, job['first_failure_date']) + MIN_DELTA_BEFORE_DEACTIVATION < now
            ):
                job['active'] = False
                job['failures'] = 0
                job['first_failure_date'] = None
                ir_cron._notify_admin_deactivation(job['cron_name'], job['id'], job['ir_actions_server_id'])
            else:
```